src/input/cursor.c

```diff
@@ -347,15 +363,31 @@ cursor_set(struct seat *seat, enum lab_cursors cursor)
 		return;
 	}
 
-	wlr_cursor_set_xcursor(seat->cursor, seat->xcursor_manager,
-		cursor_names[cursor]);
 	seat->server_cursor = cursor;
+	cursor_update_image(seat);
```

Comment:
@tokyo4j I have taken this over from your suggestion. `cursor_update_image()` does more than just `wlr_cursor_set_xcursor()`. This is OK and not to expensive?

Reply:
Hmm... now I think we should keep it as it is.
I thought it would be cleaner if we could centralize the logic updating the cursor image into `cursor_update_image()` and made the change without much thought, but `cursor_update_image()` seems to be intended for refreshing the cursor on output layout/scale changes and indeed it does more than just setting the cursor image. Maybe we can refactor this later.

Reply:
Should be good now.

Reply:
PS: A side effect is now that the cursor is initially hidden, even without touch, and becomes visible with the first mouse action. I like it, but this affects all setups. To revert to the current behavior we just have to initialize `cursor_visible` with `true`.

Reply:
`cursor_visible` is now initialized with `true` to keep the original behavior.